airflow/providers/databricks/hooks/databricks.py

```diff
@@ -235,21 +241,18 @@ def _get_aad_token(self, resource: str) -> str:
             attempt_num += 1
             sleep(self.retry_delay)
 
-    def _fill_aad_tokens(self, headers: dict) -> str:
+    def _fill_aad_headers(self, headers: dict) -> dict:
         """
-        Fills headers if necessary (SPN is outside of the workspace) and generates AAD token
+        Fills AAD headers if necessary (SPN is outside of the workspace)
         :param headers: dictionary with headers to fill-in
-        :return: AAD token
+        :return: dictionary with filled AAD headers
         """
-        # SP is outside of the workspace
-        if 'azure_resource_id' in self.databricks_conn.extra_dejson:
```

Comment:
I would keep this check inside the function, because it could be called by accident (in the future). maybe call it `_fill_aad_headers_if_needed`?

Reply:
What do you think if we call it `_get_aad_headers()`, which would return either empty dict or a filled dict? Also we won't need input arg `headers` in this case.

Then we could construct headers like:
```python
aad_headers = self._get_aad_headers()
headers = {**USER_AGENT_HEADER.copy(), **aad_headers}
```

Reply:
yes, I thought something like this. it's easier to use because the logic of adding headers is incorporating inside function...